app/post.go

```diff
@@ -734,18 +735,68 @@ func (a *App) GetFileInfosForPost(postId string, readFromMaster bool) ([]*model.
 	return infos, nil
 }
 
-func (a *App) GetOpenGraphMetadata(url string) *opengraph.OpenGraph {
+func (a *App) GetOpenGraphMetadata(requestURL string) *opengraph.OpenGraph {
 	og := opengraph.NewOpenGraph()
 
-	res, err := a.HTTPClient(false).Get(url)
+	res, err := a.HTTPClient(false).Get(requestURL)
 	if err != nil {
-		l4g.Error("GetOpenGraphMetadata request failed for url=%v with err=%v", url, err.Error())
+		l4g.Error("GetOpenGraphMetadata request failed for url=%v with err=%v", requestURL, err.Error())
 		return og
 	}
 	defer consumeAndClose(res)
 
 	if err := og.ProcessHTML(res.Body); err != nil {
-		l4g.Error("GetOpenGraphMetadata processing failed for url=%v with err=%v", url, err.Error())
+		l4g.Error("GetOpenGraphMetadata processing failed for url=%v with err=%v", requestURL, err.Error())
+	}
+
+	og = makeOpenGraphURLsAbsolute(og, requestURL)
+
+	return og
+}
+
+func makeOpenGraphURLsAbsolute(og *opengraph.OpenGraph, requestURL string) *opengraph.OpenGraph {
```

Comment:
I really think we should get away from writing functions that take a pointer argument, modify it, *and* return that same pointer. When I see that, the expectation should be that the return value actually exists for a reason and the argument is unmodified and a new object is returned. But that's usually not the case and I have to look through implementations to figure out what the actual behavior is.

The function should either just modify the argument and not return it, or return a new object.

Reply:
I'll get rid of the return argument. I would prefer not mutating the argument, but the library uses pointers for everything.